### ISSUE TITLE
Enhance Lint

### ISSUE THREAD

**Issue by chepsi** [droidconKE/droidconKeKotlin#277]
Lint is complaining a lot. We need to update it and fix the warnings and errors.
running `./gradlew ktlintCheck` will highlight the issues and `./gradlew ktlintFormat` will autofix some issues. we still might need to manually fix others.

**Comment by Raynafs** [comment]
Hey @chepsi. Willing to take this. Can I also implement spotless?

**Comment by chepsi** [comment]
@Raynafs Yes, you can work on this. I am pretty sure spotless is already implemented. You can check if that is the case. I have assigned the issue to you.

**Comment by Raynafs** [comment]
Hi @chepsi after running ./gradlew ktlintCheck it builds successfully. Not sure if something changed

**Comment by chepsi** [comment]
@Raynafs What about the warnings?

**Comment by Raynafs** [comment]
Do you mean the general warnings in different files of the project? @chepsi

**Comment by Raynafs** [comment]
Hi @chepsi I am facing an issue with the ktlint format. I'm not sure if that's what you meant when dealing with this project. 

The checks fail in the ci even when they are passing locally. Is that what you meant?

**Comment by chepsi** [comment]
@Raynafs the CI is working properly now.

**Comment by Raynafs** [comment]
Hi @chepsi. Thanks for this.
